web/models/user.ex

```diff
@@ -6,19 +6,47 @@ defmodule Pairmotron.User do
     field :email, :string
     field :active, :boolean
 
+    field :password, :string, virtual: true
+    field :password_confirmation, :string, virtual: true
+    field :password_hash, :string
+
     timestamps()
   end
 
-  @required_params [:name, :email]
-  @optional_params [:active]
+  @minimum_password_length 8
+  @required_params ~w(name email password password_confirmation)
+  @optional_params ~w(active)
 
   @doc """
   Builds a changeset based on the `struct` and `params`.
   """
   def changeset(struct, params \\ %{}) do
     struct
+    |> cast(params, ~w(name email), ~w(password password_confirmation active))
+    |> unique_constraint(:email)
+    |> validate_length(:password, min: @minimum_password_length)
+    |> validate_length(:password_confirmation, min: @minimum_password_length)
+    |> validate_confirmation(:password)
+    |> generate_password
+  end
+
+  def registration_changeset(struct, params \\ %{}) do
+    struct
     |> cast(params, @required_params, @optional_params)
-    |> validate_required([:name, :email])
+    |> unique_constraint(:email)
+    |> validate_length(:password, min: @minimum_password_length)
+    |> validate_length(:password_confirmation, min: @minimum_password_length)
+    |> validate_confirmation(:password)
+    |> generate_password
+  end
+
+  def generate_password(changeset) do
+    case changeset do
+      %Ecto.Changeset{valid?: true, changes: %{password: pass}} ->
+        put_change(changeset, :password_hash, Comeonin.Bcrypt.hashpwsalt(pass))
```

Comment:
That's a good call-out. My thought process in not making it its own module was admittedly partly due to the various projects/tutorials I looked at implementing this in the user model. It was also placed here partly because it modifies a changeset (and only a User changeset). Another (not as great) argument for it being here is that it really isn't a ton of logic.